File: docs/examples/python/delete_project.py
"""Delete a logging project together with all stored logs."""

import requests

BASE_URL = "http://localhost:3000/api"
TOKEN = "<token>"
PROJECT_UUID = "<project uuid>"


def delete_project() -> None:
    """Send DELETE /api/projects/<uuid>."""

    response = requests.delete(
        f"{BASE_URL}/projects/{PROJECT_UUID}",
        headers={"Authorization": f"Bearer {TOKEN}"},
        timeout=10,
    )
    response.raise_for_status()
    result = response.json()
    print(
        "Removed project, deleted logs:",
        result.get("deletedLogs"),
        "deleted ping-services:",
        result.get("deletedPingServices"),
    )


if __name__ == "__main__":
    delete_project()

File: docs/examples/typescript/deleteProject.ts
import { LoggerApiClient } from './loggerClient';

/**
 * Deletes a logging project together with its logs and ping-services.
 */
async function main(): Promise<void> {
  const client = new LoggerApiClient({ baseUrl: 'http://localhost:3000/api' });
  client.setToken('<paste token from login example>');

  const projectUuid = '<project uuid>';

  try {
    const result = await client.deleteProject(projectUuid);
    console.log('Project removed:', result);
  } catch (error) {
    console.error('Failed to delete project:', error);
  }
}

void main();
